File: Codes/RespMagnitudeFase.m
clc; clear; close all;

L = 3;
alpha = 0.7;

fs = 1000;

numerator = 1;
denominator = [1, zeros(1, L-1), -alpha];
H = tf(numerator, denominator, 1/fs);


omega = linspace(0, pi, 1024);
f = omega * fs / (2*pi);

[H_freq, w] = freqz(numerator, denominator, 1024, fs);


figure;
subplot(2,1,1);
plot(f, abs(H_freq));
title('Resposta de Magnitude |H(e^{j\omega})|');
xlabel('Frequência (kHz)');
ylabel('|H(e^{j\omega})|');
grid on;


subplot(2,1,2);
plot(f, angle(H_freq));
title('Resposta de Fase θ(e^{j\omega})');
xlabel('Frequência (kHz)');
ylabel('Fase (radianos)');
grid on;

figure;
zplane(numerator, denominator);
title('Diagrama de Polos e Zeros');
grid on;

File: python_codes/RespMagnitudeFase.py
from scipy.signal import freqz
import matplotlib.pyplot as plt
import numpy as np

L = 3
alpha = 0.7
fs = 1000

b = [1]
a = [1] + [0]*(L-1) + [-alpha]

w, H = freqz(b, a, worN=1024, fs=fs)
f = w / 1000

plt.figure()
plt.subplot(2, 1, 1)
plt.plot(f, np.abs(H))
plt.title('Magnitude |H(e^{jw})|')
plt.grid()

plt.subplot(2, 1, 2)
plt.plot(f, np.angle(H))
plt.title('Fase θ(e^{jw})')
plt.grid()
plt.show()